src/histogram.rs

```diff
@@ -245,10 +245,17 @@ impl Histogram {
                                   vec![],
                                   opts.common_opts.const_labels.clone()));
 
-        Histogram::with_desc(desc, &[])
+        Histogram::with_all(desc, &[], Some(opts))
     }
 
     fn with_desc(desc: Desc, label_values: &[&str]) -> Result<Histogram> {
+        Histogram::with_all(desc, label_values, None)
+    }
+
+    fn with_all(desc: Desc,
```

Comment:
seem strange here, the Desc is generated by Opts, so why do we still need to pass a `Option<HistogramOpts>`?
I think here we only need to pass a bucket instead. 

I think we can add the variable names in Opts, so we can use Opts in create function, no need to pass Desc.